src/model/model.nit

```diff
@@ -2018,42 +2035,66 @@ abstract class MPropDef
 	# Therefore the combination of identifiers is awful,
 	# the worst case being
 	#
-	# ~~~nitish
-	# "{mclassdef.mmodule.full_name}#{mclassdef.mclass.intro_mmodule.full_name}::{mclassdef.name}#{mproperty.intro_mclassdef.mmodule.full_name}::{mproperty.intro_mclassdef.name}::{name}"
-	# ~~~
+	#  * a property "p::m::A::x"
+	#  * redefined in a refinement of a class "q::n::B"
+	#  * in a module "r::o"
+	#  * so "r::o#q::n::B#p::m::A::x"
 	#
 	# Fortunately, the full-name is simplified when entities are repeated.
-	# The simplest form is "my_module#MyClass#my_property".
+	# For the previous case, the simplest form is "p#A#x".
 	redef var full_name is lazy do
 		var res = new FlatBuffer
-		res.append mclassdef.mmodule.full_name
-		res.append "#"
-		if not mclassdef.is_intro then
-			res.append mclassdef.mclass.intro_mmodule.full_name
-			res.append "::"
-		end
-		res.append mclassdef.name
+
+		# The first part is the mclassdef. Worst case is "r::o#q::n::B"
+		res.append mclassdef.full_name
+
 		res.append "#"
-		if mproperty.intro_mclassdef.mmodule != mclassdef.mmodule then
-			res.append mproperty.intro_mclassdef.mmodule.full_name
-			res.append "::"
-		end
-		if mclassdef.mclass != mproperty.intro_mclassdef.mclass then
-			res.append mproperty.intro_mclassdef.mclass.name
-			res.append "::"
+
+		if mclassdef.mclass == mproperty.intro_mclassdef.mclass then
+			# intro are unabigous in a class
+			res.append name
+		else
+			# Just try to simplify each part
+			if mclassdef.mmodule.mproject != mproperty.intro_mclassdef.mmodule.mproject then
+				# precise "p::m" only if "p" != "r"
+				res.append mproperty.intro_mclassdef.mmodule.full_name
+				res.append "::"
+			else if mproperty.visibility <= private_visibility then
+				# Same project ("p"=="q"), but private visibility,
+				# does the module part ("::m") need to be displayed
+				if mclassdef.mmodule.namespace_for(mclassdef.mclass.visibility) != mproperty.intro_mclassdef.mmodule.mproject then
+					res.append "::"
+					res.append mproperty.intro_mclassdef.mmodule.name
+					res.append "::"
+				end
+			end
+			if mclassdef.mclass != mproperty.intro_mclassdef.mclass then
+				# precise "B" only if not the same class than "A"
+				res.append mproperty.intro_mclassdef.name
+				res.append "::"
+			end
+			# Always use "x"
```

Comment:
Always use the "x" char? Or are you referring to the "x" property used in the example shown 40 lines before?